### PR TITLE
ci: change to use GitHub actions

### PR DESCRIPTION
**Pull request by FrankYang0529**
#### Which issue(s) this PR fixes:
https://github.com/longhorn/longhorn/issues/8039

### CODE REVIEW

**Comment by FrankYang0529** [comment]
Hi @derekbit, may you help to check the error test case? It looks like we didn't have CI for this repository.

**Comment by derekbit** [comment]
```
iscsi_test.go:81:
    c.Assert(err, IsNil)
... value *errors.withStack = failed to execute: /usr/bin/nsenter [nsenter iscsiadm --version], output , stderr nsenter: failed to execute iscsiadm: No such file or directory
: exit status 1 ("failed to execute: /usr/bin/nsenter [nsenter iscsiadm --version], output , stderr nsenter: failed to execute iscsiadm: No such file or directory\n: exit status 1")

```

iSCSI service is not installed.
Check the installation in the [doc](https://longhorn.io/docs/1.6.1/deploy/install/#installing-open-iscsi) on different OSes.

**Comment by derekbit** [comment]
Found found the Dockerfile.dapper relise ubuntu:16.04 base image. @FrankYang0529 Can you help create a ticket for migrating it to BCI image? Thank you.

**Comment by derekbit** [comment]
```
value *errors.withStack = failed to execute: /usr/bin/nsenter [nsenter iscsiadm -m discovery -t sendtargets -p 172.17.0.2], output , stderr iscsiadm: c
```
kernel module is missing. Try `modprobe iscsi_tcp`.

File: Dockerfile.dapper [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:16.04
+FROM registry.suse.com/bci/bci-base:15.6
```
Comment (innobead) [inline Dockerfile.dapper:1]: Use golang 1.22 bci, and remove manual go installation.

**Review by innobead — CHANGES_REQUESTED** [review]
Need to fix the failed PR check.

```console
Running tests
modprobe: FATAL: Module iscsi_tcp not found in directory /lib/modules/6.5.0-1018-azure
Failed to load iscsi_tcp module
time="2024-04-30T06:37:02Z" level=fatal msg="exit status 1"
make: *** [Makefile:11: ci] Error 1
```

File: Dockerfile.dapper [hunk 2]
```diff
@@ -29,20 +35,21 @@ ENV GOLANG_ARCH_amd64=amd64 GOLANG_ARCH_arm64=arm64 GOLANG_ARCH=GOLANG_ARCH_${AR
...
 
 # Build TGT
 RUN cd /usr/src && \
     git clone https://github.com/rancher/tgt.git && \
     cd tgt && \
     git checkout 3a8bc4823b5390e046f7aa8231ed262c0365c42c && \
-    ./scripts/build-pkg.sh deb && \
-    dpkg -i ./pkg/tgt_*.deb
+    make; \
+    make install
```
[the remark this answers is not in the document]
Comment (FrankYang0529) [reply Dockerfile.dapper:52]: Because we didn't run Drone in this repository 😅

**Comment by derekbit** [comment]
@FrankYang0529 Is it ready to review?

**Comment by FrankYang0529** [comment]
> @FrankYang0529 Is it ready to review?

This one is like go-spdk-helper and longhorn-spdk-engine. It need system modules like iSCSI daemon. Do you know how to start it in container? The testing error message is:

```
: exit status 12 ("failed to execute: /usr/bin/nsenter [nsenter iscsiadm -m discovery -t sendtargets -p 172.17.0.2], output , stderr Running in chroot, ignoring command 'start'\niscsiadm: can not connect to iSCSI daemon (111)!\niscsiadm: Could not scan /sys/class/iscsi_transport.\nlibkmod: kmod_module_insert_module: could not find module by name='iscsi_tcp'\niscsiadm: Could not insert module tcp. Kmod error -2\niscsiadm: iSCSI driver tcp is not loaded. Load the module then retry the command.\niscsiadm: Could not perform SendTargets discovery: iSCSI driver not found. Please make sure it is loaded, and retry the operation\n: exit status 12")
```

**Comment by derekbit** [comment]
> > @FrankYang0529 Is it ready to review?
> 
> This one is like go-spdk-helper and longhorn-spdk-engine. It need system modules like iSCSI daemon. Do you know how to start it in container? The testing error message is:
> 
> ```
> : exit status 12 ("failed to execute: /usr/bin/nsenter [nsenter iscsiadm -m discovery -t sendtargets -p 172.17.0.2], output , stderr Running in chroot, ignoring command 'start'\niscsiadm: can not connect to iSCSI daemon (111)!\niscsiadm: Could not scan /sys/class/iscsi_transport.\nlibkmod: kmod_module_insert_module: could not find module by name='iscsi_tcp'\niscsiadm: Could not insert module tcp. Kmod error -2\niscsiadm: iSCSI driver tcp is not loaded. Load the module then retry the command.\niscsiadm: Could not perform SendTargets discovery: iSCSI driver not found. Please make sure it is loaded, and retry the operation\n: exit status 12")
> ```

It seems kernel module is not found.
Can you try `modprobe iscsi_tcp`?

**Comment by FrankYang0529** [comment]
> > > @FrankYang0529 Is it ready to review?
> > 
> > 
> > This one is like go-spdk-helper and longhorn-spdk-engine. It need system modules like iSCSI daemon. Do you know how to start it in container? The testing error message is:
> > ```
> > : exit status 12 ("failed to execute: /usr/bin/nsenter [nsenter iscsiadm -m discovery -t sendtargets -p 172.17.0.2], output , stderr Running in chroot, ignoring command 'start'\niscsiadm: can not connect to iSCSI daemon (111)!\niscsiadm: Could not scan /sys/class/iscsi_transport.\nlibkmod: kmod_module_insert_module: could not find module by name='iscsi_tcp'\niscsiadm: Could not insert module tcp. Kmod error -2\niscsiadm: iSCSI driver tcp is not loaded. Load the module then retry the command.\niscsiadm: Could not perform SendTargets discovery: iSCSI driver not found. Please make sure it is loaded, and retry the operation\n: exit status 12")
> > ```
> 
> It seems kernel module is not found. Can you try `modprobe iscsi_tcp`?

We can't use `modprobe` command in container. It's like issue in longhorn-spdk-engine and go-spdk-helper.

**Comment by derekbit** [comment]
> > > > @FrankYang0529 Is it ready to review?
> > > 
> > > 
> > > This one is like go-spdk-helper and longhorn-spdk-engine. It need system modules like iSCSI daemon. Do you know how to start it in container? The testing error message is:
> > > ```
> > > : exit status 12 ("failed to execute: /usr/bin/nsenter [nsenter iscsiadm -m discovery -t sendtargets -p 172.17.0.2], output , stderr Running in chroot, ignoring command 'start'\niscsiadm: can not connect to iSCSI daemon (111)!\niscsiadm: Could not scan /sys/class/iscsi_transport.\nlibkmod: kmod_module_insert_module: could not find module by name='iscsi_tcp'\niscsiadm: Could not insert module tcp. Kmod error -2\niscsiadm: iSCSI driver tcp is not loaded. Load the module then retry the command.\niscsiadm: Could not perform SendTargets discovery: iSCSI driver not found. Please make sure it is loaded, and retry the operation\n: exit status 12")
> > > ```
> > 
> > 
> > It seems kernel module is not found. Can you try `modprobe iscsi_tcp`?
> 
> We can't use `modprobe` command in container. It's like issue in longhorn-spdk-engine and go-spdk-helper.

You can modprobe it in the hostname space instead.

**Comment by derekbit** [comment]
> Hi @derekbit, may you help to check the error test case? It looks like we didn't have CI for this repository.

You are right. I just realized we never run CI here.

**Comment by FrankYang0529** [comment]
> I think we can disable CI test as before and fix it later. WDYT?

Agree. The go-iscsi-helper works well. We just need to find a way to test it in the future. Thank you.

**Pull request merged**